### PR TITLE
Skip predictions in reports if prediction package is missing.

### PR DESCRIPTION
**Pull request by stkerr**
# Description

This PR does two main changes:
* It adds logic to detect if the prediction package is installed and skips all predictions steps if it is not.
* It explicitly opens the report file as UTF-8, which helps with OS like Windows.

This enables the report to still be generated and used, even if no prediction capabilities are present.

# How has this been tested?

I set up a local build with no prediction packages installed and with the feature flag disabled. I then ran the report with `reports/equity TSLA` and several other tickers. I confirmed that the reports generated successfully and that the prediction tab had a message about the prediction capabilities being missing.

# Checklist:

- [x] Update [our Hugo documentation](https://openbb-finance.github.io/OpenBBTerminal/) following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/website).
- [x] Update our tests following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/tests).
- [x] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/blob/main/CONTRIBUTING.md).
- [x] If a feature was added make sure to add it to the corresponding [scripts file](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/scripts).

# Others
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My code passes all the checks pylint, flake8, black, ... To speed up development you should run `pre-commit install`.
- [x] New and existing unit tests pass locally with my changes. You can test this locally using `pytest tests/...`.

### CODE REVIEW

**Comment by stkerr** [comment]
@jmaslek following up to the [bug](https://github.com/OpenBB-finance/OpenBBTerminal/issues/1898) I opened yesterday - this PR adds logic to check if the prediction module is present and disables the steps that need it if it is not present.

Can you review this PR? Thanks!

File: openbb_terminal/reports/equity.ipynb [hunk 1]
```diff
@@ -33,7 +33,17 @@
...
+    "\n",
+    "# Detect if prediction capabilities are present. If they are not, disable prediction in the rest of the script\n",
+    "# so that the report can still be generated without prediction results.\n",
+    "predictions = True\n",
+    "try:\n",
+    "    openbb.stocks.pred.models\n",
+    "except Exception as e:\n",
+    "    predictions = False\n",
+    "    from openbb_terminal.rich_config import console, MenuText\n",
+    "    console.print(\"Failed to import predictions package. Please ensure the prediction package is installed and the feature flag is enabled.\", \"\\n\")"
```
Comment (jmaslek) [inline openbb_terminal/reports/equity.ipynb:46]: To do a quick test, I uninstalled tensor flow and re-ran the equity report.  This line didn't actually trigger (namely I don't see this text in the terminal), but the rest ran without throwing any errors.

Screen Shot 2022-06-02 at 5 02 56 PM
s

**Comment by jmaslek** [comment]
> @jmaslek following up to the [bug](https://github.com/OpenBB-finance/OpenBBTerminal/issues/1898) I opened yesterday - this PR adds logic to check if the prediction module is present and disables the steps that need it if it is not present.
> 
> Can you review this PR? Thanks!

Thanks @stkerr!  Great catch here.  I tested it and the console.print doesn't print to the terminal (which presumably makes sense since its through paper mill).  But the report runs without TF installed, so this bug is squashed.

@martinb-bb and @piiq : we should improve the pred api so that certain models are still available in certain cases.  For example, the mc techniques are just numpy arrays so it should be usable even if tf is not installed.

**Comment by martinb-bb** [comment]
That's a great idea @jmaslek! Will look into this.

File: openbb_terminal/reports/equity.ipynb (hunk 1 above)
Comment (stkerr) [reply openbb_terminal/reports/equity.ipynb:46]: @jmaslek when this notebook runs, it looks like all output is hidden behind a progress/loading bar. I believe it should show up fine if the notebook is done directly though. That's probably not the main use case though, so let's strike these lines for now.
```suggestion
```

**Review by piiq — APPROVED** [review]

**Pull request merged**